pkg/services/authn/authnimpl/sync/oauth_token_sync.go

```diff
@@ -102,3 +103,27 @@ func (s *OAuthTokenSync) SyncOauthTokenHook(ctx context.Context, identity *authn
 
 	return nil
 }
+
+func getOAuthTokenCacheTTL(token *oauth2.Token) time.Duration {
+	ttl := maxOAuthTokenCacheTTL
+	if token == nil {
+		return ttl
+	}
+
+	if !token.Expiry.IsZero() {
+		d := time.Until(token.Expiry)
+		if d < ttl {
+			ttl = d
+		}
+	}
+
+	idTokenExpiry, err := oauthtoken.GetIDTokenExpiry(token)
+	if err == nil && !idTokenExpiry.IsZero() {
+		d := time.Until(idTokenExpiry)
+		if d < ttl {
+			ttl = d
+		}
+	}
```

Comment:
these are currently not adding any skew like we do in needTokenRefresh, it should use the same skew calculation